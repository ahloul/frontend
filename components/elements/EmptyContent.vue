<template>
  <div
    class="animated flex content-center flex-wrap justify-center h-full text-center w-full border-2 border-dark rounded-lg border-dashed text-warning mb-4"
    :class="[
      route || emit
        ? 'cursor-pointer hover:border-warning hover:text-warning'
        : '',
    ]"
    @click="action"
  >
    <div class="p-4">{{ content }}<slot /></div>
  </div>
</template>

<script>
export default {
  name: 'EmptyContent',
  props: {
    content: {
      type: String,
      default: null,
    },
    route: {
      type: String,
      default: null,
    },
    emit: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    action() {
      if (this.route) {
        this.$router.push(this.route)
      }
      if (this.emit) {
        this.$emit('action')
      }
    },
  },
}
</script>

<style lang="css" scoped></style>
