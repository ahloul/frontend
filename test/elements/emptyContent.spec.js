import EmptyContent from '../../components/elements/EmptyContent.vue'
import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

//const localvue = createLocalVue()
//localVue.use(VueRouter)
//const router = new VueRouter()

//shallowMount(EmptyContent, {
//  localVue
//})

describe('EmptyContent ', () => {
  let wrapper

  test('is a Vue instance', () => {
    wrapper = mount(EmptyContent)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('click on div', () => {
    const goToRoute = jest.fn()
    wrapper = mount(EmptyContent, {
      propsData: {
        goToRoute,
      },
    })
    wrapper.find('div').trigger('click')
    expect(goToRoute).toHaveBeenCalled
  })

  test('display content', () => {
    const content = 'Kein Content vorhanden'
    wrapper = mount(EmptyContent, {
      propsData: {
        content: content,
      },
    })
    //console.log(wrapper.html())
    expect(wrapper.find('div').text()).toBe('Kein Content vorhanden')
  })
})
