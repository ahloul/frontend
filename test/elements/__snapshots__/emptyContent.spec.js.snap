// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyContent  html should render correctly 1`] = `
"<div class=\\"animated flex content-center flex-wrap justify-center h-full text-center w-full border-2 rounded-lg border-dashed text-dark mb-4\\">
  <div class=\\"p-4\\">Kein Content vorhanden</div>
</div>"
`;
